app/scripts/metamask-controller.test.js

```diff
@@ -4528,9 +4530,12 @@ describe('MetaMaskController', () => {
         metamaskController.keyringController.state.keyrings[0].metadata.id;
 
       const wallet = {
-        discoverAndCreateAccounts: jest
+        discoverAccounts: jest
           .fn()
-          .mockResolvedValue({ Bitcoin: 1, Solana: 2 }),
+          .mockResolvedValue([
+            { type: SolAccountType.DataAccount },
+            { type: SolAccountType.DataAccount },
```

Comment:
I'd just add a EVM account too, just to make sure we only consider Solana accounts:

```suggestion
            { type: SolAccountType.DataAccount },
            { type: EthAccountType.Eoa },
```